docs/managing-environments.mdx

```diff
@@ -0,0 +1,269 @@
+
+<CodeGroup>
+```text Good Resume Prompts
+"Work in the fancy-mallard environment and add user authentication"
+
+"Continue in environment fancy-mallard - the tests are failing, please fix them"
+
+"Using the fancy-mallard environment, add CSS styling to make it look modern"
+```
+
+```text What Happens
+The agent will:
+✅ Reuse the existing container state
+✅ Have access to all previous work
+✅ Continue from where it left off
+✅ Maintain the same dependencies and setup
+```
+</CodeGroup>
+
+## Practical Examples
+
+### Example 1: Happy Path Workflow
+
+```bash
+# 1. Agent creates environment and builds feature
+$ cu list
+ID            TITLE                    CREATED       UPDATED
+fancy-mallard Flask App with Login     2 mins ago    30 secs ago
+
+# 2. Quick check - looks good!
+$ cu diff fancy-mallard
++def login():
++    # Authentication logic
++    pass
+
+# 3. Accept the work
+$ cu merge fancy-mallard
+Updating abc123..def456
+Fast-forward
+ app.py | 15 +++++++++++++++
+ 1 file changed, 15 insertions(+)
+```
+
+### Example 2: Iteration Workflow
+
+```bash
+# 1. Check what agent built
+$ cu log fancy-mallard
+commit def456 (container-use/fancy-mallard)
+Author: Agent <agent@container-use.dev>
+Date:   Mon Jan 15 10:30:00 2024 -0800
+
+    Add basic login form
+    
+Notes (container-use):
+    $ flask run
+
+# 2. Needs improvement - continue in same environment
+# Prompt: "Work in fancy-mallard environment and add password validation"
+
+# 3. Check again after agent works
+$ cu diff fancy-mallard
+# Now shows both original work + new validation
+```
+
+### Example 3: Recovery Workflow
+
+```bash
+# 1. Agent's approach isn't working
+$ cu log problematic-env
+# Shows agent went down wrong path with complex dependency issues
+
+# 2. Cut losses and start fresh  
+$ cu delete problematic-env
+
+# 3. Try different approach
+# New prompt: "Create a simple REST API using FastAPI instead of Flask"
+# Agent creates fresh environment from clean state
+```
+
+## Managing Multiple Environments
+
+You can have multiple agents working simultaneously:
+
+```bash
+$ cu list
+ID              TITLE                     CREATED       UPDATED
+frontend-work   React UI Components       5 mins ago    1 min ago
+backend-api     FastAPI User Service      3 mins ago    2 mins ago  
+data-pipeline   ETL Processing Script     1 min ago     30 secs ago
+```
+
+Each environment is completely isolated - no conflicts, no interference.
+
+## Best Practices
+
+<AccordionGroup>
+  <Accordion title="Start with Quick Assessment">
+    Always use `cu diff` and `cu log` first. Most of the time, this gives you enough information to decide next steps without the overhead of checking out or entering containers.
+  </Accordion>
+
+  <Accordion title="Merge Early and Often">
+    When an agent produces good work, merge it quickly. This creates safe checkpoints you can return to. Better to have many small merges than one large risky change.
+  </Accordion>
```

Comment:
are the env branches not safe places the user can return to? i feel like one of the benefits we offer is that you can yolo more aggressively because container-use provides best-in-class git hygiene with 0 user intervention. 

without container-use, i commit super frequently so I can reliably reset the agent to known good states. with container-use, those states appear by magic.

tbf, this is hard to express in words... like we obviously don't wanna encourage people to yolo 10000 line changes into maintained repositories for somebody else to review, but container-use does make it safer to make larger iterations